### ISSUE TITLE
deleteItems() should check items in database before deleteById()

### ISSUE THREAD

**Issue by avoidik** [skilion/onedrive#211]
This is related to 957b034805985d40a25d8521996791c74c055a64

Bug description.
deleteItems() function in sync.d (line 358) deletes items in database (line 363) without checking their existence first. This behavior throws an exception.

https://github.com/skilion/onedrive/blob/master/src/sync.d#L358
https://github.com/skilion/onedrive/blob/master/src/sync.d#L363

Error log
```
Jul 02 11:55:21 nb-lin onedrive[1870]: object.Exception@src/itemdb.d(231): Unknow item id
Jul 02 11:55:21 nb-lin onedrive[1870]: ----------------
Jul 02 11:55:21 nb-lin onedrive[1870]: /home/vja/.local/dlang/dmd-2.074.1/linux/bin64/../../src/phobos/std/exception.d:388 pure @safe bool std.exception.enforce!(Exception, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x58fbe1]
Jul 02 11:55:21 nb-lin onedrive[1870]: src/itemdb.d:231 immutable(char)[] itemdb.ItemDatabase.computePath(const(char)[], const(char)[]) [0x5c674b]
Jul 02 11:55:21 nb-lin onedrive[1870]: src/sync.d:362 void sync.SyncEngine.deleteItems() [0x5db8d7]
Jul 02 11:55:21 nb-lin onedrive[1870]: src/sync.d:185 void sync.SyncEngine.applyDifferences() [0x5d9f83]
```

**Comment by avoidik** [comment]
Same problem in here #208

**Comment by avoidik** [comment]
Some problem in here #200

**Comment by avoidik** [comment]
I guess something like this would help
```
			Item item;
			if (itemdb.selectById(i[0], i[1], item)) {
				itemdb.deleteById(i[0], i[1]);
			}
```

**Comment by skilion** [comment]
Thanks a lot for the extensive debug.
Fixed in 849dd36276f99af87b36d7541bba79ba314d5b93
